src/modules/projects/components/tables/ProjectHouseholdsTable.tsx

```diff
@@ -215,6 +345,36 @@ const ProjectHouseholdsTable = ({
       }}
       queryDocument={GetProjectHouseholdsDocument}
       columns={columns || defaultColumns}
+      overrideTableBody={true}
+      renderRow={(household, columnKeys) => {
+        return (
+          <>
+            <TableBody
+              // TODO @martha - discuss these TableBodies to achieve the `rowgroup` role, is it worth it?
```

Comment:
Using the `role='rowgroup'` turned out to be a bit complicated because of MUI expectations around usage of the `TableBody`, `TableRow` and `TableCell` components. We can't directly have `renderRow` return a `<div role='rowgroup'>` which contains all the `TableRow`s within it, because MUI expects that `TableRow` is a direct child of `TableBody`. (Here's a screenshot of what happens:)
![Screenshot 2024-11-22 at 12 28 25 PM](https://github.com/user-attachments/assets/bc59223b-dff2-4fb7-b58b-c743b474108d)

Rendering each household as its own `tbody` is the best solution I was able to come up with, but I'm not a huge fan -- both because of the messy markup and because of bloating the already complicated `GenericTable` api. Let me know if you can think of any better solutions. Or, we could skip using `rowgroup`. I think the accessibility is still better than before. We could, for example, give the row header cell additional accessible text specifying the household (HoH name or household ID) that would be read when navigating the screen with a screenreader.

Reply:
using several `tbody`s seems like a fine approach. from [mdn](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/tbody#usage_notes),

> It's permitted to use more than one <tbody> per table as long as they are all consecutive. This allows to divide the rows ([<tr>](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/tr) elements) in large tables into sections, each of which may be separately formatted if so desired.

That seems like basically what we are doing.

`overrideTableBody` prop maybe isn't the clearest as to what it's doing, since it's really omitting the `tbody` tag entirely, not overriding it.  Maybe another idea is passing `TableBodyComponent={React.Fragment}` where the default is `TableBodyComponent = TableBody`, then the passed component can be used directly? Either seems OK.

Reply:
Ok, thanks for validating the approach! I like your api suggestion. 👍

Reply:
Ok, thank you for validating the overall approach! I like your api suggestion, took a stab at implementing that here: https://github.com/greenriver/hmis-frontend/pull/989/commits/3586e74ace3065fa5d782fe4b68a12a6106d67d1